cocos/base/CCScheduler.cpp

```diff
@@ -547,10 +548,16 @@ void Scheduler::removeUpdateFromHash(struct _listEntry *entry)
     HASH_FIND_PTR(_hashForUpdates, &entry->target, element);
     if (element)
     {
-        // list entry
-        DL_DELETE(*element->list, element->entry);
-        CC_SAFE_DELETE(element->entry);
-
+        if (_updateHashLocked)
+        {
+            element->entry->markedForDeletion = true;
+        }
+        else
+        {
+            // list entry
+            DL_DELETE(*element->list, element->entry);
+            CC_SAFE_DELETE(element->entry);
+        }
```

Comment:
@minggo 
Then how about removing `Scheduler::removeUpdateFromHash` and insert that logic into `Scheduler::unscheduleUpdate`? and _updateHashLocked will be used in `Scheduler::unscheduleUpdate` again.

because
1. removeUpdateFromHash is appeared only in one place.
2. removeUpdateFromHash actually also remove item from some doubly linked list.

thanks. sorry for bad english skill.